neuralforecast/tsdataset.py

```diff
@@ -370,12 +408,178 @@ def __init__(
+        )
+
+        return item
+
+    @staticmethod
+    def from_data_directories(
+        directories,
+        static_df=None,
+        sort_df=False,
+        exogs=[],
+        id_col="unique_id",
+        time_col="ds",
+        target_col="y",
+    ):
+        """We expect directories to be a list of directories of the form [unique_id=id_0, unique_id=id_1, ...]. Each directory should contain the timeseries corresponding to that unqiue_id,
+        represented as a pandas or polars DataFrame. The timeseries can be entirely contained in one parquet file or split between multiple, but within each parquet files the timeseries should be sorted by time.
+        Static df should also be a pandas or polars DataFrame"""
+        import pyarrow as pa
+
+        # Define indices if not given and then extract static features
+        static, static_cols = TimeSeriesDataset._extract_static_features(
+            static_df, sort_df, id_col
+        )
+
+        max_size = 0
+        min_size = float("inf")
+        last_times = []
+        ids = []
+        exogs = set(exogs)
+        # Check if available_mask is present in all the directories
+        available_mask_seen = False
+
+        for dir in directories:
+            dir_path = Path(dir)
+            if not dir_path.is_dir():
+                raise ValueError(f"paths must be directories, {dir} is not.")
+            uid = dir_path.name.split("=")[-1]
+            total_rows = 0
+            last_time = None
+            for file in dir_path.glob("*.parquet"):
+                meta = pa.parquet.read_metadata(file)
+                rg = meta.row_group(0)
+                col2pos = {
+                    rg.column(i).path_in_schema: i for i in range(rg.num_columns)
+                }
+
+                last_time_file = (
+                    meta.row_group(meta.num_row_groups - 1)
+                    .column(col2pos[time_col])
+                    .statistics.max
+                )
+                last_time = (
+                    max(last_time, last_time_file)
+                    if last_time is not None
+                    else last_time_file
+                )
+                total_rows += sum(
+                    meta.row_group(i).num_rows for i in range(meta.num_row_groups)
+                )
+
+                # Check all the temporal columns are present
+                missing_cols = exogs - col2pos.keys()
+                if missing_cols:
+                    raise ValueError(
+                        f"Temporal columns: {missing_cols} not found in the file: {file}."
+                    )
+
+                if available_mask_seen and ("available_mask" not in col2pos.keys()):
+                    raise ValueError(
+                        f"available_mask column not found in the file: {file}, but found in previous files"
+                    )
+                elif "available_mask" in col2pos.keys():
+                    available_mask_seen = True
```

Comment:
This will add it the first time it's seen and will check it in the following iterations.
```suggestion
                if "available_mask" in col2pos:
                    expected_temporal.add("available_mask")
```

Reply:
Sounds good! I have also included the edge case that we see a number of files that don't contain the available_mask column before a file with the available_mask column present, as I don't think we want to add available_mask to temporal_cols in this case

```
if "available_mask" not in col2pos.keys():
        available_mask_seen = False
elif not available_mask_seen:
        # If this is triggered the available_mask column is present in this file but has been missing from previous files.
        raise ValueError("The available_mask column is present in some files but is missing in others." )
else:
        expected_temporal.add("available_mask")
```

Reply:
Awesome!